src/ImageSharp/Formats/IImageDecoder.cs

```diff
@@ -24,6 +24,16 @@ public interface IImageDecoder
         Image<TPixel> Decode<TPixel>(Configuration configuration, Stream stream)
             where TPixel : unmanaged, IPixel<TPixel>;
 
+        /// <summary>
+        /// Decodes the image from the specified stream to an <see cref="Image{TPixel}"/> of a specific pixel type.
+        /// </summary>
+        /// <typeparam name="TPixel">The pixel format.</typeparam>
+        /// <param name="stream">The <see cref="Stream"/> containing image data.</param>
+        /// <param name="image">The target <see cref="Image{TPixel}"/> instance.</param>
+        // TODO: Document ImageFormatExceptions (https://github.com/SixLabors/ImageSharp/issues/1110)
+        void Decode<TPixel>(Stream stream, Image<TPixel> image)
```

Comment:
This is gonna be a breaking change for users who implemented their `IImageDecoder`. The reason @DaZombieKiller proposed `IImageMemoryDecoder` was to avoid this.

I'm undecided on this, with a slight preference to not break.

Reply:
Hmmmm.... It's a fairly fundamental break and there are decoder out in the wild. I say new interface.

Reply:
In that case, now comes the hard part: naming it. `IImageMemoryDecoder` doesn't seem descriptive enough, especially since it's no longer operating on `Memory<T>` like in the original proposal. Maybe `IImageTargetDecoder` or `IImageTargetedDecoder`, based on the name of this PR?

Reply:
We'd have to expose the interface and use the type as the method parameter type. Can't call what you don't have then.

Reply:
What about the overloads that don't take an `IImageDecoder` currently? Would they just be removed?

Reply:
When someone calls the new `Image.Load` without a specified decoder we will loop through our config to find a matching decoder that implements the interface. If there is none it will through. 

I generally don't recommend people use specific decoders unless they have to as they skip the format validation checks.